### PR TITLE
document specifying your own renovate config for the nudging

### PR DESCRIPTION
**Pull request by rcerven**
STONEBLD-2916

### CODE REVIEW

File: docs/modules/ROOT/pages/how-tos/configuring/component-nudges.adoc [hunk 1]
```diff
@@ -109,3 +109,23 @@ NOTE: You can also use **kubectl edit** to add the build-nudges-ref instead of p
...
 * Try running the necessary commands again, and ensure you have the correct syntax.
+
+== Using custom renovate config
+It is possible to configure your own custom renovate config, which will be used during nudging.
+Renovate config has to be specified in ConfigMap in your namespace.
+
+.There are two possible options:
+. create global config in ConfigMap named `global-nudging-renovate-config` which will be used for all nudged components in your namespace,
+unless component specific config exists
+. create config for specific nudged component in ConfigMap named `nudging-renovate-config-COMPONENT_NAME` and
```
Comment (mmorhun) [inline docs/modules/ROOT/pages/how-tos/configuring/component-nudges.adoc:120]: Usually `<component-name>` style is used

File: docs/modules/ROOT/pages/how-tos/configuring/component-nudges.adoc [hunk 2]
```diff
@@ -109,3 +109,23 @@ NOTE: You can also use **kubectl edit** to add the build-nudges-ref instead of p
...
+it will be used for the component in your namespace
```
Comment (mmorhun) [inline docs/modules/ROOT/pages/how-tos/configuring/component-nudges.adoc:121]: Missing `.`

File: docs/modules/ROOT/pages/how-tos/configuring/component-nudges.adoc [hunk 3]
```diff
@@ -109,3 +109,23 @@ NOTE: You can also use **kubectl edit** to add the build-nudges-ref instead of p
...
+
+Both ConfigMaps have the same format, can contain `config.json` or `config.js` keys, if both specified `config.json` has precedence.
+In the value of the key is js or json in the string.
+
+When you have either of ConfigMaps, your config will be used unmodified and it is users responsibility to fix it when renovate pipeline will be failing because of it.
```
Comment (mmorhun) [inline docs/modules/ROOT/pages/how-tos/configuring/component-nudges.adoc:126]: `users` -> `user's`

File: docs/modules/ROOT/pages/how-tos/configuring/component-nudges.adoc [hunk 4]
```diff
@@ -109,3 +109,23 @@ NOTE: You can also use **kubectl edit** to add the build-nudges-ref instead of p
...
+
+.Examples
```
Comment (mmorhun) [inline docs/modules/ROOT/pages/how-tos/configuring/component-nudges.adoc:128]: Can we have more comprehensive example with auto-merging and/or commit message changed as requested by users.

File: docs/modules/ROOT/pages/how-tos/configuring/component-nudges.adoc [hunk 5]
```diff
@@ -109,3 +109,23 @@ NOTE: You can also use **kubectl edit** to add the build-nudges-ref instead of p
 
 * Ensure that the existing references in the repositories of your components are correct.
 * Try running the necessary commands again, and ensure you have the correct syntax.
+
+== Using custom renovate config
```
Comment (mmorhun) [inline docs/modules/ROOT/pages/how-tos/configuring/component-nudges.adoc:113]: It's really not clear how renovate is related to nudges. We should explain it here.

File: docs/modules/ROOT/pages/how-tos/configuring/component-nudges.adoc (hunk 1 above)
Comment (rcerven) [reply docs/modules/ROOT/pages/how-tos/configuring/component-nudges.adoc:120]: ok

File: docs/modules/ROOT/pages/how-tos/configuring/component-nudges.adoc (hunk 2 above)
Comment (rcerven) [reply docs/modules/ROOT/pages/how-tos/configuring/component-nudges.adoc:121]: ok

File: docs/modules/ROOT/pages/how-tos/configuring/component-nudges.adoc (hunk 3 above)
Comment (rcerven) [reply docs/modules/ROOT/pages/how-tos/configuring/component-nudges.adoc:126]: ok

File: docs/modules/ROOT/pages/how-tos/configuring/component-nudges.adoc (hunk 4 above)
Comment (rcerven) [reply docs/modules/ROOT/pages/how-tos/configuring/component-nudges.adoc:128]: examples are only to show the formatting

for specific options, users should consult renovate docs

File: docs/modules/ROOT/pages/how-tos/configuring/component-nudges.adoc (hunk 5 above)
Comment (rcerven) [reply docs/modules/ROOT/pages/how-tos/configuring/component-nudges.adoc:113]: ok

**Review by mmorhun — APPROVED** [review]

File: docs/modules/ROOT/pages/how-tos/configuring/component-nudges.adoc [hunk 6]
```diff
@@ -109,3 +109,24 @@ NOTE: You can also use **kubectl edit** to add the build-nudges-ref instead of p
...
 * Ensure that the existing references in the repositories of your components are correct.
 * Try running the necessary commands again, and ensure you have the correct syntax.
+
+== Customizing nudging PRs
+Nudging feature is using renovate `https://docs.renovatebot.com/` to actually do the job,
+for which you can supply your own renovate config, which will be used during nudging.
+Renovate config has to be specified in ConfigMap in your namespace.
+
+.There are two possible options:
+. Create namespace wide config in ConfigMap named `namespace-wide-nudging-renovate-config` which will be used for all nudged components in your namespace,
```
Comment (arewm) [inline docs/modules/ROOT/pages/how-tos/configuring/component-nudges.adoc:119]: Do Konflux users have the permissions to create a ConfigMap in their workspace? If so, what role do they need?

Comment (rcerven) [reply docs/modules/ROOT/pages/how-tos/configuring/component-nudges.adoc:119]: I can create it in my own namespace, and from UI I can see that it is granted by role:
appstudio-admin-user-actions

Comment (rcerven) [reply docs/modules/ROOT/pages/how-tos/configuring/component-nudges.adoc:119]: yeah seems like namespace admins can only create it, which is fine I'd say

Comment (rcerven) [reply docs/modules/ROOT/pages/how-tos/configuring/component-nudges.adoc:119]: mmm I can add it, but it depends on infra/workspace configuration ....

**Review by arewm — APPROVED** [review]

**Pull request merged**